src/server/qgsserverfeatureid.h

```diff
@@ -0,0 +1,64 @@
+/***************************************************************************
+                              qgsserverfeatureid.h
+                              -----------------------
+  begin                : May 17, 2019
+  copyright            : (C) 2019 by René-Luc DHONT
+  email                : rldhont at 3liz dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSSERVERFEATUREID_H
+#define QGSSERVERFEATUREID_H
+
+#include <QString>
+#include <QHash>
+
+#include "qgis_server.h"
+#include "qgsfield.h"
+
+class QgsVectorDataProvider;
+class QgsFeature;
+class QgsFeatureRequest;
+
+#ifdef SIP_RUN
+% ModuleHeaderCode
+#include "qgsserverfeatureid.h"
+% End
+#endif
+
+
+/**
+ * \ingroup server
+ * The QgsServerFeatureId namespace provides a way to use primary keys for
+ * feature id.
+ * \since QGIS 3.4.8
+ */
+namespace QgsServerFeatureId
+{
+
+  /**
+   * Returns the feature id based on primary keys.
+   * \param f the feature
+   * \param pkAttributes the primary keys list
+   * \returns the feature id based on primary keys
+   * \since QGIS 3.4.8
+   */
+  SERVER_EXPORT QString getFeatureId( const QgsFeature *f, const QgsAttributeList &pkAttributes );
+
+  SERVER_EXPORT QgsFeatureRequest updateFeatureRequestFromFeatureIds( QgsFeatureRequest &featureRequest, const QStringList &fids, const QgsVectorDataProvider *provider );
```

Comment:
Modifying in place, or returning an updated copy?

Reply:
Modifying in place